x-pack/plugins/security_solution/cypress/e2e/investigations/alerts/expandable_flyout/alert_details_right_panel_overview_tab.cy.ts

```diff
@@ -98,7 +98,7 @@ describe(
 
         cy.get(DOCUMENT_DETAILS_FLYOUT_OVERVIEW_TAB_DESCRIPTION_TITLE)
           .should('be.visible')
-          .and('have.text', 'Rule description');
+          .and('have.text', 'Rule descriptionView rule');
```

Comment:
I feel like I would prefer keeping this test just for the actual title, so sticking with only 'Rule description', like

```typescript
.and('contain.text', 'Rule description');
```

then I would had another check to verify the existence of the button. As you can see [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/timelines/components/timeline/body/renderers/formatted_field_helpers.tsx#L126) we pass down `ruleName` to the [LinkButton component](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/common/components/links/helpers.tsx#L70) we're displaying.

So you could add a variable like

```typescript
export const DOCUMENT_DETAILS_FLYOUT_OVERVIEW_TAB_NAVIGATE_TO_RULE_DETAILS_BUTTON = getDataTestSubjectSelector('ruleName');
```

to our [alert_details_right_panel_overview_tab.cy.ts](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/cypress/screens/expandable_flyout/alert_details_right_panel_overview_tab.ts) file

Then use it here like

```typescript
cy.get(DOCUMENT_DETAILS_FLYOUT_OVERVIEW_TAB_DESCRIPTION_TITLE)
    .should('be.visible')
    .within(() => {
        cy.get(DOCUMENT_DETAILS_FLYOUT_OVERVIEW_TAB_NAVIGATE_TO_RULE_DETAILS_BUTTON)
            .should('be.visible')
            .and('have.text', 'View rule');
    });
```

To push things even further, we could check that clicking on the button does navigate to the rule details page  but that might be a bit overkill, and would mean that we either navigate pack to the alerts page and reopen the flyout before continuing the test, or have a separate test for that...

Reply:
good call, thanks for pointing this out!